src/NuGet.Core/NuGet.Packaging/ContentModel/ContentItemCollection.cs

```diff
@@ -198,7 +196,7 @@ public ContentItemGroup FindBestItemGroup(SelectionCriteria criteria, params Pat
             return null;
         }
 
-        private IEnumerable<ContentItem> FindItemsImplementation(PatternSet definition, IEnumerable<Asset> assets)
+        private void FindItemsImplementation(PatternSet definition, IEnumerable<Asset> assets, IList<ContentItem> itemsList)
```

Comment:
Is the item list reused anywhere?

If we just change the return type from IEnumerable to List, are the savings comparable?

Reply:
This would require a much more significant change in the logic of the code. For the benefits to be given with minimal modification, a translation from the yield return that allocates an enumerator each call to instead supplying a list that will be added to allows the logic to remain the same but the allocation happens only by the caller.

Reply:
This was in my "Review started" for more than 11 days apparently.

Reply:
I see what you mean. I can repurpose the existing private method to return a list instead of do a yield return. That will work for both the new API and the old one.